pkg/installmanager/dnscleanup.go

```diff
@@ -63,3 +67,35 @@ func cleanupAWSDNSZone(dnsZone *hivev1.DNSZone, region string, logger log.FieldL
 	zoneLogger.Info("DNSZone cleaned")
 	return nil
 }
+
+func cleanupGCPDNSZone(dnsZone *hivev1.DNSZone, logger log.FieldLogger) error {
+	if dnsZone.Status.GCP == nil {
+		return fmt.Errorf("found non-GCP DNSZone for DNS ClusterDeployment")
+	}
+	if dnsZone.Status.GCP.ZoneName == nil {
+		// Shouldn't happen as we block installs until DNS is ready
+		return fmt.Errorf("DNSZone %s has no ZoneName set", dnsZone.Name)
+	}
+
+	zoneLogger := logger.WithField("zoneName", *dnsZone.Status.GCP.ZoneName)
```

Comment:
Just change `logger`. Otherwise, you need to also change all of the other uses of `logger` in this function to use `zoneLogger` instead.
```suggestion
	logger = logger.WithField("zoneName", *dnsZone.Status.GCP.ZoneName)
```